pkg/allocation/converters/converter_test.go

```diff
@@ -910,8 +912,9 @@ func TestConvertGSAToAllocationResponse(t *testing.T) {
 					NodeName: "node-name",
 				},
 			},
-			wantErrCode:      codes.ResourceExhausted,
-			skipConvertToGSA: true,
+			grpcUnallocatedStatusCode: codes.ResourceExhausted,
+			wantErrCode:               codes.ResourceExhausted,
+			skipConvertToGSA:          true,
 		},
```

Comment:
Add another table entry here that tests the non-default case:

```
		{
			name: "status field is set to unallocated, non-default unallocated",
			in: &allocationv1.GameServerAllocation{
				TypeMeta: metav1.TypeMeta{
					Kind:       "GameServerAllocation",
					APIVersion: "allocation.agones.dev/v1",
				},
				Status: allocationv1.GameServerAllocationStatus{
					State:          allocationv1.GameServerAllocationUnAllocated,
					GameServerName: "GSN",
					Ports: []agonesv1.GameServerStatusPort{
						{
							Port: 123,
						},
						{
							Name: "port-name",
						},
					},
					Address:  "address",
					NodeName: "node-name",
				},
			},
			grpcUnallocatedStatusCode: codes.Unimplemented,
			wantErrCode:               codes.Unimplemented,
			skipConvertToGSA:          true,
		},
```

This is basically just a permutation of the test right above, but with a new unallocated status. I chose `codes.Unimplemented` because it's different than existing codes we might return for other reasons, making it easier to see, if this test were to fail.

Reply:
I was trying with the following test case, it wasn't working for me.🤔

{
    name: "Test non-default error code",
    in: &allocationv1.GameServerAllocation{
        TypeMeta: metav1.TypeMeta{
            Kind:       "GameServerAllocation",
            APIVersion: "allocation.agones.dev/v1",
        },
        Status: allocationv1.GameServerAllocationStatus{
            State:          allocationv1.GameServerAllocationUnAllocated,
            GameServerName: "RestrictedAccessGSN",
            Ports: []agonesv1.GameServerStatusPort{
                {
                    Port: 456,
                    Name: "restricted-port-name",
                },
            },
            Address:  "restricted-address",
            NodeName: "restricted-node-name",
        },
    },
    grpcUnallocatedStatusCode: codes.PermissionDenied,
    want: &pb.AllocationResponse{
        GameServerName: "RestrictedAccessGSN",
        Address:        "restricted-address",
        NodeName:       "restricted-node-name",
        Ports: []*pb.AllocationResponse_GameServerStatusPort{
            {
                Port: 456,
                Name: "restricted-port-name",
            },
        },
        Metadata: &pb.AllocationResponse_GameServerMetadata{
            Labels: map[string]string{
                "error": "permission-denied",
            },
        },
    },
    wantErrCode: codes.PermissionDenied,
},